File: frontend/src/types/vehicle.ts
/** 车辆相关类型定义 */

export interface Vehicle {
  id: number
  plateNumber: string
  coachId: number
  coachName: string
  vehicleType: string
  status: 'NORMAL' | 'MAINTENANCE' | 'STOPPED'
  remark: string
  createTime: string
}

export interface VehicleSimple {
  id: number
  plateNumber: string
  status: string
}

export interface VehicleCreateDTO {
  plateNumber: string
  coachId?: number
  vehicleType?: string
  remark?: string
}

export interface VehicleUpdateDTO {
  plateNumber: string
  coachId?: number
  vehicleType?: string
  remark?: string
}

export interface VehicleQuery {
  page?: number
  size?: number
  keyword?: string
  coachId?: number
  status?: string
}

File: backend/src/main/java/com/drivingschool/vehicle/entity/Vehicle.java
package com.drivingschool.vehicle.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 车辆实体 — 对应 vehicle 表
 * <p>
 * 车辆可绑定默认教练（coach_id），排班时默认使用该教练的车辆。
 * 维修中或停用的车辆不可被排班绑定。
 * </p>
 */
@Data
@TableName("vehicle")
public class Vehicle {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 车牌号，唯一 */
    private String plateNumber;

    /** 默认绑定教练ID */
    private Long coachId;

    /** 车型 */
    private String vehicleType;

    /** 状态：NORMAL / MAINTENANCE / STOPPED */
    private String status;

    /** 备注 */
    private String remark;

    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
